src/mosquito/rate_limiter.cr

```diff
@@ -0,0 +1,135 @@
+module Mosquito::RateLimiter
+  module ClassMethods
+    # Configures rate limiting for this job.
+    #
+    # `limit` and `per` are used to control the run count and the window
+    # duration. Defaults to a limit of 1 run per second.
+    #
+    # `increment` is used to indicate how many "hits" against a single job is
+    # worth. Defaults to 1.
+    #
+    # `key` is used to combine rate limiting functions across multiple jobs.
+    def throttle(
+      limit : Int32 = 1,
+      per : Time::Span = 1.second,
+      increment = 1,
+      key = self.name.underscore
```

Comment:
What are your thoughts on requiring named args here? `throttle(*, limit, per...)`

Reply:
Good catch. I think I intended for that, but just left out the leading `*`.